source/vibe/http/server.d

```diff
@@ -707,28 +707,53 @@ final class HTTPServerRequest : HTTPRequest {
 	@property URL fullURL()
 	const {
 		URL url;
-		auto fh = this.headers.get("X-Forwarded-Host", "");
-		if (!fh.empty) {
-			url.schema = this.headers.get("X-Forwarded-Proto", "http");
-			url.host = fh;
+
+		// Set URL host segment.
+		if (auto xfh = this.headers.get("X-Forwarded-Host")) {
+			url.host = xfh;
+		} else if (!this.host.empty) {
+			url.host = this.host;
+		} else if (!m_settings.hostName.empty) {
+			url.host = m_settings.hostName;
 		} else {
-			if (!this.host.empty) url.host = this.host;
-			else if (!m_settings.hostName.empty) url.host = m_settings.hostName;
-			else url.host = m_settings.bindAddresses[0];
+			url.host = m_settings.bindAddresses[0];
+		}
+
+		// Set URL schema segment.
+		if (auto xfp = this.headers.get("X-Forwarded-Proto")) {
+			url.schema = xfp;
+		} else if (this.tls) {
+			url.schema = "https";
+		} else {
+			url.schema = "http";
+		}
 
-			if (this.tls) {
-				url.schema = "https";
-				if (m_port != 443) url.port = m_port;
+		// Set URL port segment.
+		Nullable!ushort port;
+		if (auto xfp = this.headers.get("X-Forwarded-Port")) {
+			try {
+				port = xfp.to!ushort;
+			} catch (ConvException) {
+				// Value was not ushort.
```

Comment:
Another possibility would be to respond with a 400 error here. May be the better option in case of e.g. proxy configuration errors. But I'd say that there should be a diagnostic log message here at least.